job-server/src/spark.jobserver/JobManager.scala

```diff
@@ -44,12 +44,18 @@ object JobManager {
     } else {
       defaultConfig
     }
+
+    val system = makeSystem(config.resolve())
+    val clazz = Class.forName(config.getString("spark.jobserver.jobdao"))
+    val ctor = clazz.getDeclaredConstructor(Class.forName("com.typesafe.config.Config"))
+    val jobDAO = ctor.newInstance(config).asInstanceOf[JobDAO]
```

Comment:
The only change I am making here is to pass jobDAO. I am not sure what you mean by moved to an object.